litr/src/main/java/com/linkedin/android/litr/filter/video/gl/ScaleToFitGlFrameRenderFilter.java

```diff
@@ -0,0 +1,172 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// from: https://android.googlesource.com/platform/cts/+/lollipop-release/tests/tests/media/src/android/media/cts/TextureRender.java
+// blob: 4125dcfcfed6ed7fddba5b71d657dec0d433da6a
+// modified: removed unused method bodies
+// modified: use GL_LINEAR for GL_TEXTURE_MIN_FILTER to improve quality.
+// modified: added filters
+package com.linkedin.android.litr.filter.video.gl;
+
+import android.opengl.GLES11Ext;
+import android.opengl.GLES20;
+import android.opengl.Matrix;
+
+import androidx.annotation.NonNull;
+
+import com.linkedin.android.litr.filter.GlFrameRenderFilter;
+import com.linkedin.android.litr.render.GlRenderUtils;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.FloatBuffer;
+
+public class ScaleToFitGlFrameRenderFilter implements GlFrameRenderFilter {
+
+    private static final int FLOAT_SIZE_BYTES = 4;
+    private static final int TRIANGLE_VERTICES_DATA_STRIDE_BYTES = 5 * FLOAT_SIZE_BYTES;
+    private static final int TRIANGLE_VERTICES_DATA_POS_OFFSET = 0;
+    private static final int TRIANGLE_VERTICES_DATA_UV_OFFSET = 3;
+
+    // shaders
+    private static final String VERTEX_SHADER =
+            "uniform mat4 uMVPMatrix;\n" +
+                    "uniform mat4 uSTMatrix;\n" +
+                    "attribute vec4 aPosition;\n" +
+                    "attribute vec4 aTextureCoord;\n" +
+                    "varying vec2 vTextureCoord;\n" +
+                    "void main() {\n" +
+                    "  gl_Position = uMVPMatrix * aPosition;\n" +
+                    "  vTextureCoord = (uSTMatrix * aTextureCoord).xy;\n" +
+                    "}\n";
+    private static final String FRAGMENT_SHADER =
+            "#extension GL_OES_EGL_image_external : require\n" +
+                    "precision mediump float;\n" +      // highp here doesn't seem to matter
+                    "varying vec2 vTextureCoord;\n" +
+                    "uniform samplerExternalOES sTexture;\n" +
+                    "void main() {\n" +
+                    "  gl_FragColor = texture2D(sTexture, vTextureCoord);\n" +
+                    "}\n";
+
+    private FloatBuffer triangleVertices;
+    private final float[] triangleVerticesData = {
+            // X, Y, Z, U, V
+            -1.0f, -1.0f, 0, 0.f, 0.f,
+            1.0f, -1.0f, 0, 1.f, 0.f,
+            -1.0f, 1.0f, 0, 0.f, 1.f,
+            1.0f, 1.0f, 0, 1.f, 1.f,
+    };
+
+    private float[] mvpMatrix = new float[16];
+    private float[] inputTextureTransformMatrix = new float[16];
+
+    private int glProgram;
+    private int mvpMatrixHandle;
+    private int uStMatrixHandle;
+    private int aPositionHandle;
+    private int aTextureHandle;
+    private int inputTextureId;
```

Comment:
Keep the same coding style and sequence of the original open source might make later maintenance/upgrade easier. Thought?

Reply:
Good call. Done.

Reply:
If check the original file https://android.googlesource.com/platform/cts/+/lollipop-release/tests/tests/media/src/android/media/cts/TextureRender.java, you will find mvpMatrixHandle was from muMVPMatrixHandle, uStMatrixHandle was from muSTMatrixHandle, etc. I am think if the two files can be diff easily, and patch any fix easily. But let you decide.